src/main/kotlin/com/ampnet/identityservice/service/impl/UserServiceImpl.kt

```diff
@@ -41,16 +47,41 @@ class UserServiceImpl(
         return user
     }
 
-    @Transactional
-    override fun updateEmail(email: String, address: String): User = getUser(address).apply { this.email = email }
-
     @Transactional
     override fun createUser(address: String): User =
         userRepository.findByAddress(address) ?: kotlin.run {
             logger.info { "User is created for address: $address" }
             userRepository.save(User(address))
         }
 
+    @Transactional
+    override fun updateEmail(email: String, address: String): User {
+        val user = getUser(address)
+        val mailToken = MailToken(0, address, email, UUID.randomUUID(), ZonedDateTime.now())
```

Comment:
It's better to use provider services for `UUID` and current date-time because then it is more testable as you don't have any side-effects in your code. Defining those provider services is quite simple, so I would suggest doing it.

```kotlin
// interfaces which are easily mockable in tests
interface UuidProvider {
    fun getUuid(): UUID
}

interface ZonedDateTimeProvider {
    fun getZonedDateTime(): ZonedDateTime
}

@Service
class RandomUuidProvider {
    override fun getUuid() = UUID.randomUUID()
}

@Service
class CurrentZonedDateTimeProvider {
    override fun getZonedDateTime() = ZonedDateTime.now()
}
```